src/web/nextui/src/app/eval/ResultsView.tsx

```diff
@@ -222,44 +242,55 @@ export default function ResultsView({
         }`,
         group: 'Variables',
       })),
-      ...head.prompts.map((_, idx) => {
-        const prompt = head.prompts[idx];
-        const label = prompt.label || prompt.display || prompt.raw;
-        return {
-          value: `Prompt ${idx + 1}`,
-          label: `Prompt ${idx + 1}: ${label.length > 100 ? label.slice(0, 97) + '...' : label}`,
-          group: 'Prompts',
-        };
-      }),
+      ...promptOptions,
     ];
-  }, [head.vars, head.prompts, hasAnyDescriptions]);
+  }, [head.vars, promptOptions, hasAnyDescriptions]);
 
   const [configModalOpen, setConfigModalOpen] = React.useState(false);
   const [viewSettingsModalOpen, setViewSettingsModalOpen] = React.useState(false);
-  const [columnVisibility, setColumnVisibility] = React.useState<VisibilityState>({});
-  const [selectedColumns, setSelectedColumns] = React.useState<string[]>(
-    columnData.map((col) => col.value),
-  );
 
-  const handleChange = (event: SelectChangeEvent<typeof selectedColumns>) => {
-    const {
-      target: { value },
-    } = event;
-    setSelectedColumns(typeof value === 'string' ? value.split(',') : value);
-
-    const allColumns = [
+  const allColumns = useMemo(
+    () => [
       ...(hasAnyDescriptions ? ['description'] : []),
       ...head.vars.map((_, idx) => `Variable ${idx + 1}`),
       ...head.prompts.map((_, idx) => `Prompt ${idx + 1}`),
-    ];
-    const newColumnVisibility: VisibilityState = {};
-    allColumns.forEach((col) => {
-      newColumnVisibility[col] = (typeof value === 'string' ? value.split(',') : value).includes(
-        col,
-      );
-    });
-    setColumnVisibility(newColumnVisibility);
-  };
+    ],
+    [hasAnyDescriptions, head.vars, head.prompts],
+  );
+
+  const updateColumnVisibility = useCallback(
+    (columns: string[]) => {
+      const newColumnVisibility: VisibilityState = {};
+      allColumns.forEach((col) => {
+        newColumnVisibility[col] = columns.includes(col);
+      });
+      setColumnState(currentEvalId, {
+        selectedColumns: columns,
+        columnVisibility: newColumnVisibility,
+      });
+    },
+    [allColumns, setColumnState, currentEvalId],
+  );
+
+  useEffect(() => {
```

Comment:
nit: maybe move `currentColumnState` definition down here to above where it's used?